server/mailet/rate-limiter-redis/src/main/scala/org/apache/james/rate/limiter/redis/RedisRateLimiter.scala

```diff
@@ -67,6 +67,10 @@ class RedisRateLimiterFactory @Inject()(redisConfiguration: RedisConfiguration)
       masterReplicaRedisConfiguration.redisURI.value.asJava,
       masterReplicaRedisConfiguration.readFrom)
 
+    case sentinelRedisConfiguration: SentinelRedisConfiguration =>
+      sentinelRedisConfiguration.maybeSentinelPassword.foreach(password => sentinelRedisConfiguration.redisURI.getSentinels.forEach(uri => uri.setPassword(password)))
+      new RedisSingleInstanceRateLimitjFactory(RedisClient.create(sentinelRedisConfiguration.redisURI))
```

Comment:
`RedisSingleInstanceRateLimitjFactory`

Do you test case when 1 sentinel node down, then client will auto switch to another sentinel node?
(My concern, because the classname is "SingleInstance")

Reply:
No, I don't have test for that. All communications to sentinel are handled by lettuce and the class name has nothing to do here.

Reply:
Test added